### PR TITLE
Support Psycopg3 synchronous operations

### LINKED ISSUE #1789: Instrument psycopg3
psycopg3 was released some time ago. We currently only support psycopg2. While psycopg3 is quite similar from psycopg2, the differences are large enough to probably require its own instrumentation (e.g., extensions are gone in psycopg3 AFAICT).

Also, there is an asyncio interface in psycopg3, which requires its own instrumentation.

https://www.psycopg.org/psycopg3/docs/index.html
https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html

### PR DESCRIPTION
**Pull request by minhtule**
## What does this pull request do?

Support Psycopg3 synchronous operations by instrumenting [`psycopg.connect`](https://www.psycopg.org/psycopg3/docs/api/module.html#psycopg.connect). Add `PsycopgInstrumentation` which adopts [`Psycopg2Instrumentation`](https://github.com/elastic/apm-agent-python/blob/main/elasticapm/instrumentation/packages/psycopg2.py#L77).
- In naming, use `psycopg` (not `psycopg3`) because Psycopg3's module name is [`psycopg`](https://pypi.org/project/psycopg/).
- This PR doesn't support Psycopg3 [asynchronous operations](https://www.psycopg.org/psycopg3/docs/advanced/async.html).

## Related issues

Closes #1789

### CODE REVIEW

File: tests/instrumentation/psycopg_tests.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+        assert span["name"] == "SELECT FROM test"
+        assert "db" in span["context"]
+        assert span["context"]["db"]["instance"] == "elasticapm_test"
+        assert span["context"]["db"]["type"] == "sql"
+        assert span["context"]["db"]["statement"] == baked_query
+
+
+@pytest.mark.integrationtest
+@pytest.mark.skipif(not has_postgres_configured, reason="PostgresSQL not configured")
+def test_psycopg_call_stored_function(instrument, postgres_connection, elasticapm_client):
```
Comment (minhtule) [inline tests/instrumentation/psycopg_tests.py:199]: `psycopg2_tests.py` has [a similar test](https://github.com/minhtule/apm-agent-python/blob/35acac150833e5b451a82a276d892dd6f6c75c28/tests/instrumentation/psycopg2_tests.py#L427-L447) to test [`cursor.callproc()`](https://www.psycopg.org/docs/cursor.html#cursor.callproc). However, Psycopg3 removes this method and recommends to use `cursor.execute()` ([doc](https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html#callproc-is-gone)) so I'm not really sure if we should keep this test.

Comment (basepi) [reply tests/instrumentation/psycopg_tests.py:199]: I think that's reasonable, we can skip that test.

**Review by basepi — APPROVED** [review]
This looks great! I'm going to get @beniwohli to take a look as well and then we can get this merged!

File: tests/instrumentation/psycopg_tests.py (hunk 1 above)
Comment (minhtule) [reply tests/instrumentation/psycopg_tests.py]: Thank you for confirming! I've removed this test.

**Review by beniwohli — APPROVED** [review]
Awesome!

**Comment by minhtule** [comment]
@basepi @beniwohli Thank you very much for your quick response and review! 🙏

**Pull request merged**